docs/benchmarking.rst

```diff
@@ -0,0 +1,259 @@
+preceding the exclamation mark is otherwise interpreted as an escape character.
+In most other shells (e.g. on Linux or macOS) the literal quotation marks are unnecessary.
+
+To run only benchmarks from a certain example, we can
+use the `--bench` argument, which allows us to write a "regular expression" that
+filters the benchmarks actually executed.  This is handy if you are interested in
+benchmarking a particular model or component, as running *all* the benchmarks can
+take a very long time, and the larger benchmarks (e.g. on the full SANDAG model)
+will need a lot of disk space and RAM.  For example, to run only the mandatory
+tour frequency benchmark for the SANDAG 1-Zone example-sized system, run::
+
+    activitysim benchmark latest --bench sandag1example.time_mandatory_tour_frequency
+
+The "." character here means a literal dot, but since this is a regex expression,
+it is also a single-character wildcard.  Thus, you can run all the example-sized
+SANDAG benchmarks with::
+
+    activitysim benchmark latest --bench sandag.example
+
+You can also repeat the `--bench` argument to give multiple different expressions.
+So, you can run just the 1- and 2-zone examples, without the 3-zone example::
+
+    activitysim benchmark latest --bench sandag1example --bench sandag2example
+
+
+Threading Limits
+~~~~~~~~~~~~~~~~
+
+When you run benchmarking using the `activitysim benchmark` command, the
+following environment variable are set automatically before benchmarking begins::
+
+    MKL_NUM_THREADS = 1
+    OMP_NUM_THREADS = 1
+    OPENBLAS_NUM_THREADS = 1
+    NUMBA_NUM_THREADS = 1
+    VECLIB_MAXIMUM_THREADS = 1
+    NUMEXPR_NUM_THREADS = 1
+
+This ensures that all benchmarking operations run processes in single-threaded
+mode.  This still allows ActivitySim itself to spin up multiple processes if the
+item being timed is a multiprocess benchmark.
+
+Submitting Benchmarks
+~~~~~~~~~~~~~~~~~~~~~
+
+One of the useful features of the airspeed velocity benchmarking engine is the
+opportunity to compare performance benchmarks across different machines. The
+ActivitySim community is interested in aggregating such results from a number
+of participants, so once you have successfully run a set of benchmarks, you
+should submit those results to our repository.
+
+To do so, assuming you have run the benchmark tool inside the ``asim-benchmarks``
+repository as noted above, you simply need to commit any new or changed files
+in the ``asim-benchmarks/results`` directory.  You can then open a pull request
+against the community ``asim-benchmarks`` to submit those results.
+
+Assuming you are in (or first ``cd`` into) the ``asim-benchmarks`` directory, You can
+do this from the command line using the following steps::
+
+    git add results
+    pre-commit run    # required on Windows only, see note
+    git commit -m "adding benchmark results"
+    git push
+    gh pr create
```

Comment:
Looks ok, except your local machine paths were in the commit.  I've been trying to figure out how to manage that automatically (unsuccessfully so far).  It's not a big deal so I don't want to block the ASV work because of it.